### PR TITLE
refactor(core): refactor error handler logic

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

This PR includes the following update:

1. Delete the original OIDCRequestError type
2. Introduce new koa-oidc-error-handler map `OIDCProviderError` to `RequestError`
3. Introduce new koa-slonik-error-handler map `SlonikError` to `RequestError`
4. Add optional table name and id field to the Slonik DeletionError, help us to track the error context
5. Add the optional expose field to the RequestErrorMetadata
5. Remove the old OIDC and Slonik related error handle logic from the koa-error-handler

## Linear Issue Reference

LOG1486

## Testing

Test locally on the local server
image

@logto-io/eng

### CODE REVIEW

File: packages/core/src/middleware/koa-slonik-error-handler.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      if (!(error instanceof SlonikError)) {
+        throw error;
+      }
+
+      switch (error.constructor) {
+        case DeletionError:
+        case NotFoundError:
+          throw new RequestError({
+            code: 'entity.not_found',
+            status: 404,
```
Comment (wangsijie) [inline packages/core/src/middleware/koa-slonik-error-handler.ts:43]: wondering if there is a convenient way to change status code in some situations.

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/middleware/koa-slonik-error-handler.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:43]: Do you mean to change the original staus code of Slonik error types?

Comment (wangsijie) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:43]: what if I want to change status (404) to 422

Comment (simeng-li) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:43]: hmm so far the error handler itself only catches the Original Slonik error (without any status code), at the very end of the request flow.  In your case, the only thing we have is to catch the error at the route level and return a customized RequestError instead?  Either way, the error needs to be cached at the route level and passed to the handler.

**Pull request merged**